extensions/vscode/src/state.ts

```diff
@@ -81,8 +81,13 @@ interface extensionContext {
   readonly workspaceState: Memento;
 }
 
+export interface CredentialRefreshEvent {
+  readonly oldCredentials: Credential[];
+}
+
 export class PublisherState implements Disposable {
   private readonly context: extensionContext;
+  private credentialRefresh = new EventEmitter<CredentialRefreshEvent>();
```

Comment:
Looks like you'll need to add `EventEmitter` to our mock for vscode (extensions/vscode/src/state.test.ts, line 64), in order to get the existing unit tests to work.